sklearn/linear_model/tests/test_quantile.py

```diff
@@ -143,22 +147,23 @@ def test_asymmetric_error(quantile):
     )
     intercept = 1.23
     coef = np.array([0.5, -2])
+    assert np.max(X @ coef + intercept) > 0, "take care that X @ coef + intercept > 0"
     # For an exponential distribution with rate lambda, e.g. exp(-lambda * x),
     # the quantile at level q is:
     #   quantile(q) = - log(1 - q) / lambda
-    #   scale = 1/lambda = -quantile(q) / log(1-q)
+    #   scale = 1/lambda = -quantile(q) / log(1 - q)
     y = rng.exponential(
         scale=-(X @ coef + intercept) / np.log(1 - quantile), size=n_samples
     )
     model = QuantileRegressor(
         quantile=quantile,
         alpha=0,
-        solver="interior-point",
-        solver_options={"tol": 1e-5},
+        solver="highs",
     ).fit(X, y)
+
     assert model.intercept_ == approx(intercept, rel=0.2)
     assert_allclose(model.coef_, coef, rtol=0.6)
-    assert_allclose(np.mean(model.predict(X) > y), quantile)
+    assert_allclose(np.mean(model.predict(X) > y), quantile, rtol=6e-3)
```

Comment:
Does it helpt to play with `solver_options={"dual_feasibility_tolerancedouble": 1e-7, "primal_feasibility_tolerance": 1e-7, "ipm_optimality_tolerance": 1e-8}` in order to reduce the `rtol` a little? Maybe to `1e-3`?

Reply:
On the other hand, with `n_samples=1000` a tolerance smaller than `1e-2` is ok.

Reply:
See https://github.com/scikit-learn/scikit-learn/pull/21546#discussion_r763223837